include/model/CInterimBucketCorrector.h

```diff
@@ -48,40 +50,39 @@ class MODEL_EXPORT CInterimBucketCorrector {
     //! Constructs an interim bucket corrector for buckets of length \p bucketLength
     CInterimBucketCorrector(core_t::TTime bucketLength);
 
-    //! Copy constructor
-    CInterimBucketCorrector(const CInterimBucketCorrector& other);
+    //! Updates the current count in the bucket.
+    void currentBucketCount(core_t::TTime time, std::uint64_t count);
+
+    //! Updates the bucket count model with a new bucket count.
+    void finalBucketCount(core_t::TTime time, std::uint64_t bucketCount);
 
-    //! Updates the model of the bucket count with a new measurement
-    void update(core_t::TTime time, std::size_t bucketCount);
+    //! Get an estimate of the current bucket completeness.
+    double completeness() const;
```

Comment:
In case of latency this will return the completeness of the last bucket for which `estimateBucketCompleteness` is called, right?

Reply:
That's right. This is just a cache of the result of completeness calculation which is populated in the `CCountingModel::sampleBucketStatistics`. We should call all these sample methods before using the completeness estimate and these are always called when getting interim results so this really just to avoid us calling `estimateBucketCompleteness` for each individual model, just to compute the same value.

This is actually a convenient optimisation, because I noticed that evaluating the trend cropped up high when profiling recently, i.e. #73.